src/main/java/seedu/task/model/task/DescContainsKeywordsPredicate.java

```diff
@@ -16,7 +16,8 @@ public DescContainsKeywordsPredicate(String keyphrase) {
 
     @Override
     public boolean test(Task task) {
-        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase);
+        return StringUtil.containsIgnoreCase(task.getDescription().value, keyphrase)
+                && task.getDescription().getHasDescription();
```

Comment:
Would it be good to allow the user to search for tasks that have no descriptions too?

Reply:
Good for future consideration, will implement if time permits.